ios/Mobile/MobileSocket.mm

```diff
@@ -0,0 +1,145 @@
+// -*- Mode: objc; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4; fill-column: 100 -*-
+/*
+ * Copyright the Collabora Online contributors.
+ *
+ * SPDX-License-Identifier: MPL-2.0
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+#import "config.h"
+
+#import "MobileSocket.h"
+
+#import "MobileApp.hpp"
+
+#import <Foundation/NSSet.h>
+#import <Foundation/NSURL.h>
+#import <Foundation/NSURLResponse.h>
+
+#import <Foundation/NSString.h>
+
+#import <Poco/URI.h>
+
+#import <wsd/DocumentBroker.hpp>
+
+@interface MobileSocket()
+{
+    dispatch_queue_t queue;
+    int serial;
+    std::vector<std::string> sendingMessages;
+    NSMutableSet<id<WKURLSchemeTask>> *ongoingTasks;
```

Comment:
I am concerned that the ongoingTasks is mutated from the main queue, and checked/mutated also from the proxySocketQueue queue, can't we have race conditions here? Like - there's addObject:/removeObject: called on the main thread in open: and write:, but also containsObject:/removeObject: run on the proxySocketQueue (in the dispatch_async). But the NSMutableSet itself is not thread-safe, is it?